### DISCUSSION TITLE
Is it possible to trigger onRecordAfterUpdate hook only by a specific row change?

### DISCUSSION THREAD

**Discussion by 0x6775737461766f** [pocketbase/pocketbase#4491] [Q&A]
For example, i have a collection "contents", and inside it there is an array field called "users", i want to send an e-mail everytime a user is appended to this relation, i saw on the documentation that onRecordAfterUpdate may help me on this, but i understand that everytime there is an update to any field in the row the trigger is activated.

**Comment by ganigeorgiev** [comment]
No, there is no such functionality at the moment.

You'll have to create a diff from the new field value and the old one.
Considering the lowercased name of the hook in your description, I assume that you use the JS `pb_hooks`, so here is one possible pseudo-code:

```js
onRecordAfterUpdateRequest((e) => {
    const oldUsers = e.record.originalCopy().get("users");
    const newUsers = e.record.get("users");

    const removedUsers = oldUsers.filter((id) => !newUsers.includes(id))
    const addedUsers = newUsers.filter((id) => !oldUsers.includes(id))

    if (!addedUsers.length && !removedUsers.length) {
        return // no users changes
    }

    // do something with addedUsers and removedUsers...
}, "contents")
```

**Reply by 0x6775737461766f** [reply]
Can i expand this relationship, given users table is a collection? For example, i know addedUsers will return me an array or single added user, but can i directly access his e-mail, given the id is returned by this?

**Reply by ganigeorgiev** [reply]
You can call `findRecordsByIds` to retrieve the user records:

```js
const users = $app.dao().findRecordsByIds("users", addedUsers)
```
For all record helpers, please refer to https://pocketbase.io/docs/js-records/.
